### PR TITLE
port:VLC : upgrade and provide private FFMpeg v2.8.6

### PR DESCRIPTION
**Pull request by RJVB**
This revision upgrades VLC to v2.2.4 and works around the FFMpeg v3.x
incompatibility issue by shipping a private, stripped-down FFMpeg 2.8.6
copy.

### CODE REVIEW

File: multimedia/VLC/Portfile [hunk 1]
```diff
@@ -1,36 +1,26 @@
...
-long_description    VLC media player is a highly portable multimedia player for \
-                    various audio and video formats (MPEG-1, MPEG-2, MPEG-4, DivX, \
-                    mp3, ogg, ...) as well as DVDs, VCDs, and various streaming protocols. \
-                    It can also be used as a server to stream in unicast or multicast \
-                    in IPv4 or IPv6 on a high-bandwidth network.
-
-homepage            http://www.videolan.org
-
-subport             lib${name} {
+# $Id: Portfile 146115 2016-02-28 16:09:05Z devans@macports.org $
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: No `$Id$` tags please, especially not expanded ones.

File: multimedia/VLC/Portfile [hunk 2]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
+    license             GPL-2
+
+    platforms           darwin
+
+    master_sites        http://download.videolan.org/pub/videolan/vlc/${version}/
+    distname            vlc-${version}
+    use_xz              yes
+
+#     checksums           rmd160  8c6d4194da37210fe409c965f430898fbe7f2416 \
+#                         sha256  b9d7587d35f13c3c981964c8cc8b03f1c7c8edf528be476b3ca1d2efedd5bf5b
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: Why is there a commented set of checksums here?

File: multimedia/VLC/Portfile [hunk 3]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
+    checksums           rmd160  2fe178bb9a4b2ad5ca0294d3e52ceb992b22e8ee \
+                        sha256  1632e91d2a0087e0ef4c3fb4c95c3c2890f7715a9d1d43ffd46329f428cf53be
+
+    if {[info exist ::env(MACPORTS_COMPRESS_WORKDIR)] && $::env(MACPORTS_COMPRESS_WORKDIR)} {
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: Please remove this code, that's certainly not something that should have special handling in the VLC port. It should either be done for all ports in base, or not at all.

File: multimedia/VLC/Portfile [hunk 4]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
-}
 
-# VLC 2.2.2 should have full Lua 5.3 compatibility, for now, depend on port:lua52
-# That requires the patch-for-lua52.diff which is appended to patchfiles below.
-# update 2.2.2 : lua 5.3 gives compiler syntax errors so we stick with 5.2 for now.
-depends_lib-append  port:lua52
-# depends_lib-append  port:lua
+    depends_build       port:pkgconfig
+
+    depends_lib         path:${FFMPEG_VLC_PREFIX}/lib/libavcodec.dylib:ffmpeg-VLC \
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: Isn't the `ffmpeg-VLC` port the only port that provides this file anyway? If so, why have the `path:`-style dependency?

File: multimedia/VLC/Portfile [hunk 5]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
+#     patchfiles-append   patch-opencv_includes.diff
+#     # The opencv example apparently builds on 10.11 but I'm having issues with it on 10.9
+#     # it doesn't appear to be indispensable, so simply skip it.
+#     patchfiles-append   patch-no-opencv_example.diff
+
+    # VLC 2.2.2 should have full Lua 5.3 compatibility, for now, depend on port:lua52
+    # That requires the patch-for-lua52.diff which is appended to patchfiles below.
+    # update 2.2.2 : lua 5.3 gives compiler syntax errors so we stick with 5.2 for now.
+    depends_lib-append  port:lua52
+    # depends_lib-append  port:lua
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: Remove the old lua dependency if it is no longer necessary. Useless comments just clutter the Portfile.

File: multimedia/VLC/Portfile [hunk 6]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
+                        PR-34741-no__clang_version__.patch \
+                        static_assert.patch \
+                        no-sparkle.patch \
+                        patch-vlc-no-O4.diff \
+                        patch-ffmpegVLC.diff
 
-compiler.blacklist  gcc-4.2 llvm-gcc-4.2 {clang < 300}
+    post-patch {
+        reinplace "s:librsvg-2/librsvg:librsvg:" \
+            ${worksrcpath}/modules/text_renderer/svg.c
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: Why not use a patchfile for this? That has the advantage of failing when the change no longer applies. The reinplace will just stay silent.

File: multimedia/VLC/Portfile [hunk 7]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
 
-pre-fetch {
-    if {${os.platform} eq "darwin" && ${os.major} < 10} {
-        ui_error "${name} ${version} requires Mac OS X 10.6 or greater."
-        return -code error "incompatible Mac OS X version"
+        if {[string match *clang* ${configure.cxx}] && ${configure.cxx_stdlib} == "libc++"} {
+            reinplace "s:-lstdc\+\+:-lc++:" \
+                ${worksrcpath}/configure.ac \
+                ${worksrcpath}/modules/access/Makefile.am
+        }
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: Could use a patchfile for this and apply it conditionally. Also, is there no way to pass those in with a variable? This seems quite hacky. 👎

File: multimedia/VLC/Portfile [hunk 8]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
+        reinplace "s|HAVE_LIBAVFORMAT_AV|HAVE_LIBAVFORMAT_VLC_AV|g" \
+                ${worksrcpath}/modules/codec/avcodec/avcommon.h \
+                ${worksrcpath}/modules/codec/avcodec/avcommon_compat.h
+        reinplace "s|HAVE_LIBAVCODEC_AV|HAVE_LIBAVCODEC_VLC_AV|g" \
+                ${worksrcpath}/modules/codec/avcodec/avcommon.h \
+                ${worksrcpath}/modules/codec/avcodec/avcommon_compat.h
+        reinplace "s|HAVE_LIBSWSCALE_SW|HAVE_LIBSWSCALE_VLC_SW|g" \
+                ${worksrcpath}/modules/codec/avcodec/avcommon.h \
+                ${worksrcpath}/modules/codec/avcodec/avcommon_compat.h
+    }
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: Ugh, that's one giant hack. Can you not prepend the include path and library path of the private copy of ffmpeg to the include and library search paths for the build of VLC instead? That would be much cleaner and avoid patching upstream source.

File: multimedia/VLC/Portfile [hunk 9]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
+    post-configure {
+        system -W ${worksrcpath} "patch -Np0 -i ${filespath}/patch-static_assert.diff"
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: Why can this not be patched using the normal `patchfiles` approach? Why is it necessary at all?

File: multimedia/VLC/Portfile [hunk 10]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
+    build.args-append   DESTDIR=${worksrcpath}/dest_ignore V=1
+
+    # gl.c:121:3: error: Platform not recognized.
+    configure.cppflags-append -D__unix__=1
+
+    # live555 is installed to a weird location
+    configure.cppflags-append -I${prefix}/lib/live/liveMedia/include
+
+    build.target        all
+    destroot.target     install
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: Those are the defaults, aren't they? Why respecify?

File: multimedia/VLC/Portfile [hunk 11]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
+            reinplace "s/extern char \\*\\*environ;/#include <crt_externs.h>/" \
+                ${worksrcpath}/modules/misc/inhibit/xdg.c \
+                ${worksrcpath}/modules/stream_filter/decomp.c
+
+            if {![file exists ${worksrcpath}/contrib/BGHUDAppKit.framework]} {
+                ln -s ${frameworks_dir}/BGHUDAppKit.framework ${worksrcpath}/contrib/BGHUDAppKit.framework
+            }
+
+            # To trick configure
+            file mkdir "${worksrcpath}/contrib/lib"
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: Please turn those reinplaces into patches, especially the `environ` -> `_NSGetEnviron()` and other environment-related ones. There no good reason for those to be reinplaces, for the reasons outlined above.

File: multimedia/VLC/Portfile [hunk 12]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
+                    # http://trac.macports.org/ticket/35131
+                    ln -s ${prefix}/lib ${destroot}${applications_dir}/VLC.app/Contents/MacOS/lib
+                    ln -s ${prefix}/lib/vlc/plugins ${destroot}${applications_dir}/VLC.app/Contents/MacOS/plugins
+                    ln -s ${prefix}/lib/vlc/lua ${destroot}${applications_dir}/VLC.app/Contents/MacOS/share/lua
+                    ln -s ${prefix}/share/locale ${destroot}${applications_dir}/VLC.app/Contents/MacOS/share/locale
+
+                    # the vlc executable needs to be started with a full path to the app bundle executable
+                    # or else the Mac OS X interface will hang beyond even a ^C or ^\ :
+                    move ${destroot}${prefix}/bin/vlc ${destroot}${prefix}/bin/vlc.exe
+                    system "echo \"#!/bin/sh\nexec \\\"${applications_dir}/VLC.app/Contents/MacOS/VLC\\\" \\\"\\\$\@\\\"\" > ${destroot}${prefix}/bin/vlc"
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: You are aware that you can just open and write a file from Tcl?

File: multimedia/VLC/Portfile [hunk 13]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
+#################  support port ffmpeg-VLC  ##################
+##
+if {${subport} eq "ffmpeg-VLC"} {
+    PortGroup           xcodeversion 1.0
 
-platform macosx {
-    default_variants-append +quartz +qtkit
+    version             2.8.6
+    revision            1
+    license             LGPL-2.1+
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: You have a duplicate license line below. Please remove the duplicate info, that's just confusing.

File: multimedia/VLC/Portfile [hunk 14]
```diff
@@ -40,434 +30,673 @@ subport             lib${name} {
...
+        }
+        # oblige dependent code to include files from our own renamed header file directories, so it
+        # cannot include mismatching headers by accident (e.g. those from ffmpeg 3.x).
+        foreach dir {libavcodec libavformat libavresample libavutil libpostproc libswresample libswscale} {
+            file rename ${destroot}${FFMPEG_VLC_PREFIX}/include/${dir} ${destroot}${FFMPEG_VLC_PREFIX}/include/${dir}-VLC
+        }
+        foreach dir {libavcodec libavformat libavresample libavutil libpostproc libswresample libswscale} {
+            foreach header [glob -nocomplain ${destroot}${FFMPEG_VLC_PREFIX}/include/*/*.h] {
+                reinplace "s|${dir}/|${dir}-VLC/|g" ${header}
+            }
```
Comment (neverpanic) [inline multimedia/VLC/Portfile]: Again, this renaming looks like a huge hack to me. Why not just configure for a separate prefix and avoid the mess?

File: multimedia/VLC/archive.sh [hunk 15]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
```
Comment (neverpanic) [inline multimedia/VLC/archive.sh]: Unused file?

File: multimedia/VLC/files/patch-static_assert.diff [hunk 16]
```diff
@@ -0,0 +1,11 @@
+--- config.h.orig    2016-03-22 15:34:33.000000000 +0000
++++ config.h 2016-03-22 15:34:57.000000000 +0000
+@@ -793,6 +793,7 @@
+ /* Define to `int' if <stddef.h> does not define. */
+ /* #undef ssize_t */
+
++#define static_assert _Static_assert
```
Comment (neverpanic) [inline multimedia/VLC/files/patch-static_assert.diff]: Why is this needed?

File: multimedia/VLC/files/patch-strverscmp.diff [hunk 17]
```diff
@@ -0,0 +1,39 @@
+diff --git compat/strverscmp.c compat/strverscmp.c
```
Comment (neverpanic) [inline multimedia/VLC/files/patch-strverscmp.diff]: I can't find a reference to `patch-strverscmp.diff`. Is this unused?

File: multimedia/VLC/files/patch-vlc-for-macports-old.diff [hunk 18]
```diff
@@ -0,0 +1,160 @@
+diff --git modules/lua/vlc.c modules/lua/vlc.c
```
Comment (neverpanic) [inline multimedia/VLC/files/patch-vlc-for-macports-old.diff]: Unused patchfile?

File: multimedia/VLC/files/patch-vlc-for-macports2.diff [hunk 19]
```diff
@@ -0,0 +1,135 @@
+diff --git src/darwin/dirs.c src/darwin/dirs.c
```
Comment (neverpanic) [inline multimedia/VLC/files/patch-vlc-for-macports2.diff]: Unused patchfile?

File: multimedia/VLC/files/patch-vlc-qt4mac.diff [hunk 20]
```diff
@@ -0,0 +1,97 @@
+diff --git modules/gui/qt4/util/searchlineedit.cpp modules/gui/qt4/util/searchlineedit.cpp
```
Comment (neverpanic) [inline multimedia/VLC/files/patch-vlc-qt4mac.diff]: Unused patchfile?

File: multimedia/VLC/files/patch-vlc-qt4mac.diff [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
++//     void updateText( const QString& );
++//     void searchEditingFinished();
++//
++// signals:
++//     void searchDelayedChanged( const QString& );
++//     void textEdited( const QString& );
++// };
+ #endif
+ 
+ #endif
```
Comment (neverpanic) [inline multimedia/VLC/files/patch-vlc-qt4mac.diff]: Why is there an empty `qtkit.patch` file below? And why does GitHub not let me add comments on empty files?!

File: multimedia/VLC/files/vlc [hunk 22]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
```
Comment (neverpanic) [inline multimedia/VLC/files/vlc]: Maybe I missed it, but it seems like the Portfile does not actually use this file at all?

File: multimedia/VLC/Portfile (hunk 3 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: I'll comply here because I don't consider VLC one of my own ports. There are times you can throw people a bone and let them do things that are somewhat off the beaten path.

File: multimedia/VLC/Portfile (hunk 7 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: That's not mine, it's been in since at least v2.1.5 .
The replacement expression is safe and as far as I'm concerned it's exactly the sort of operation for which you'd use reinplace.

File: multimedia/VLC/Portfile (hunk 8 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: Actually, that's being done through the ffmpeg-VLC .pc files. Of course I did try to make this as simple as possible, but in the end you're seeing the simplest solution that worked. IIRC part of the problem was avoiding the headerfiles from port:ffmpeg.

File: multimedia/VLC/Portfile (hunk 9 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: This is a patch that's made to a file generated during the configure step. It's been a while but from what I remember it would have been much more complicated to patch things so that the file is generated correctly.

File: multimedia/VLC/Portfile (hunk 10 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: Indeed. But you should ask the original author who added those lines.

File: multimedia/VLC/Portfile (hunk 11 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: This is another bit of code that I didn't write, and I'm frankly not very motivated to spend time to convert these into patchfiles - and keep the programmatic version around because a next VLC release is likely to break the patch but not the search/replace action...

As far as I'm concerned VLC has a build system that hangs together with hatpins and zipties, the Portfile has never been much better, and I'm only maintaining it because I need libVLC (and that already costs me more than the minimal effort I'd like to put into it).

File: multimedia/VLC/Portfile (hunk 12 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: Yes. I adopted the system echo trick from one of Mojca's contributions to my audacity port.

File: multimedia/VLC/Portfile (hunk 14 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: it is being configured that way, but without this mess the compiler will pick up the wrong headerfiles.

The mess could be avoided by configuring port:ffmpeg to put its headers elsewhere but that's not an option, and I prefer to let the portfile jump through well-tested hoops rather than obliging the user to deactivate port:ffmpeg in order to build VLC.

File: multimedia/VLC/files/patch-static_assert.diff (hunk 16 above)
Comment (RJVB) [inline multimedia/VLC/files/patch-static_assert.diff]: To make the code build on 10.10 or 10.11, I forget which.

File: multimedia/VLC/files/patch-vlc-qt4mac.diff (hunk 21 above)
Comment (RJVB) [inline multimedia/VLC/files/patch-vlc-qt4mac.diff]: The answer to the 2nd question is probably: because someone @github decided that shouldn't be allowed...

File: multimedia/VLC/files/vlc (hunk 22 above)
Comment (RJVB) [inline multimedia/VLC/files/vlc]: no you're right. This was used before I adopted Mojca's approach to create the file on the fly

File: multimedia/VLC/Portfile (hunk 6 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: With the risk of enforcing your point, this patch was apparently no longer required.

I realise that the diff makes it appear that I changed almost everything in this Portfile, but in fact all I did was move some things around and change indentation. A lot of the patching you're commenting on was there when I stepped in. As others before me I'm not really motivated to start doing a whole lot of cleaning, refactoring etc. I just change what's required to make sure the whole thing builds, that's the ultimate goal and already requires enough investment. I do improve where I can and see fit when relevant.
FWIW, I never asked or suggested to be listed as the port maintainer, I just happen to be the one who apparently cares most to keep it afloat.

File: multimedia/VLC/Portfile (hunk 4 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: I must have had a reason at some point, and never saw one to rewrite the dependency as it works exactly as intended the way it's written.

I didn't mention it, but the ffmpeg-VLC subport was designed soonish after port:ffmpeg was upgraded to 3.x, and presented on Trac. No one ever committed it, simply.

File: multimedia/VLC/Portfile (hunk 5 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: new lua, actually but indeed.

**Comment by RJVB** [comment]
I think I'm done with round 1. I haven't seen any echo of my comments appear in my email nor do I see something like a "Publish" button so I hope the comment button below will serve that purpose. New commit coming up.

**Comment by mojca** [comment]
> I haven't seen any echo of my comments appear in my email nor do I see something like a "Publish" button so I hope the comment button below will serve that purpose.

You generated "just" 18 emails in each committer's mailbox. (That's just to put your request to remove notifications of Trac attachments into proper perspective.)

**Comment by RJVB** [comment]
On Thursday November 10 2016 07:17:17 Mojca Miklavec wrote:

> You generated "just" 18 emails in each committer's mailbox. (That's just to put your request to remove notifications of Trac attachments into proper perspective.)

Ah. Sorry about that, but if we're really putting things in proper perspective: there's about just as much that can be done about that as there isn't about the trac notifications:

[quote=https://github.com/settings/notifications]
"
...
Settings
Choose which email updates you receive on conversations you're participating in or watching

 [x] Comments on Issues and Pull Requests 
"

FWIW, ReviewBoard doesn't publish each comment separately and gives the author the possibility to send an email notification or not.

R.

**Comment by mkae** [comment]
@RJVB, I am not sure, but it could be that GitHub's request review process avoids sending messages for every single comment given, if you tell it, that you want to submit a review right at the beginning of your comments.

File: multimedia/VLC/Portfile (hunk 14 above)
Comment (RJVB) [inline multimedia/VLC/Portfile]: Please also see the discussion on Trac from when I worked on this:

https://trac.macports.org/ticket/50697#comment:39

**Comment by neverpanic** [comment]
@RJVB Oh, I didn't read the old Portfile, just the new one after the changes, so it may well be that I commented on a number of things that are already in the Portfile. Feel free to ignore those comments, then.

**Comment by RJVB** [comment]
Great, thanks.

I haven't looked at the interface here too closely, but other reviewboards I know have a side-by-side view. That makes it much easier to see what is really new/changed code, and what is just whitespace change, whereas the udiff notation may make it seem much more has changed.

**Comment by favadi** [comment]
@RJVB GitHub has side-by-side view too, you have to click the "Split" button.

**Comment by RJVB** [comment]
To come back to the hacked-in support for building against ffmpeg-2 with ffmpeg-3 installed: something of the sort would also be required when following VLC's own ffmpeg-contrib build.
Most importantly though I repeat that this is a temporary hack. The next (major) VLC release is supposed to support ffmpeg-3. 
I would really prefer not to spend much more time on this version except from keeping it afloat.

**Comment by mkae** [comment]
I guess those 5 misnamed patch files
```
$ port lint --nitpick
--->  Verifying Portfile for VLC
Warning: Patchfile buildfix-package.mak.patch does not follow the source patch naming policy "patch-*.diff"
Warning: Patchfile configure.ac-no-arch.patch does not follow the source patch naming policy "patch-*.diff"
Warning: Patchfile PR-34741-no__clang_version__.patch does not follow the source patch naming policy "patch-*.diff"
Warning: Patchfile static_assert.patch does not follow the source patch naming policy "patch-*.diff"
Warning: Patchfile no-sparkle.patch does not follow the source patch naming policy "patch-*.diff"
--->  0 errors and 5 warnings found.
```
are legacy, right?

**Comment by mkae** [comment]
@RJVB, I could successfully build VLC, but I could not play an MP4 video with it:
```
VLC media player 2.2.4 Weatherwax (revision 2.2.3-37-g888b7e89)
[00007f92f17043f8] core libvlc: Running vlc with the default interface. Use 'cvlc' to use vlc without interface.
2016-12-13 05:21:02.319 VLC[27267:618996] Failed to connect (o_t4_txt_codecVideo) outlet from (VLCWizard) to (NSTextField): missing setter or instance variable
2016-12-13 05:21:02.319 VLC[27267:618996] Failed to connect (o_t4_txt_bitrateVideo) outlet from (VLCWizard) to (NSTextField): missing setter or instance variable
2016-12-13 05:21:02.319 VLC[27267:618996] Failed to connect (o_t4_txt_codecAudio) outlet from (VLCWizard) to (NSTextField): missing setter or instance variable
2016-12-13 05:21:02.319 VLC[27267:618996] Failed to connect (o_t4_txt_bitrateAudio) outlet from (VLCWizard) to (NSTextField): missing setter or instance variable
2016-12-13 05:21:02.320 VLC[27267:618996] Failed to connect (o_tc_name) outlet from (VLCPlaylistWizard) to (NSTableColumn): missing setter or instance variable
2016-12-13 05:21:02.320 VLC[27267:618996] Failed to connect (o_tc_author) outlet from (VLCPlaylistWizard) to (NSTableColumn): missing setter or instance variable
2016-12-13 05:21:02.320 VLC[27267:618996] Failed to connect (o_tc_duration) outlet from (VLCPlaylistWizard) to (NSTableColumn): missing setter or instance variable
[00007f92f3391958] macosx vout display error: no OpenGL hardware acceleration found. this can lead to slow output and unexpected results
2016-12-13 05:21:02.426 VLC[27267:618996] Can't find app with identifier com.spotify.client
2016-12-13 05:21:02.453 VLC[27267:618996] Can't find app with identifier com.spotify.client
[00007f92f3391958] macosx vout display error: Initialization of open gl view failed
[00007f92f38a40b8] core video output error: video output creation failed
[00007f92f3881eb8] core decoder error: failed to create video output
2016-12-13 05:21:02.513 VLC[27267:618996] Can't find app with identifier com.spotify.client
[00007f92f32e9e08] macosx vout display error: no OpenGL hardware acceleration found. this can lead to slow output and unexpected results
[00007f92f32e9e08] macosx vout display error: Initialization of open gl view failed
[00007f92f38a1ab8] core video output error: video output creation failed
[00007f92f3881eb8] core decoder error: failed to create video output
```
the player seemed to work as time in its progress bar progressed, but there was no windows opened for the video output.

A VOB file also could NOT be played in that "El Capitan" Parallels 11 Virtual Machine...

Alternatively I tried the MPV player and it also failed to display its video output window in that VM, so I figure that this is a VM-related problem, since it worked just fine on my Mavericks host system in MPV!

**Comment by mkae** [comment]
@neverpanic & @RJVB, I'd squash and commit this, if you don't object, okay?

**Comment by RJVB** [comment]
On Monday December 12 2016 20:06:56 Marko K�ning wrote:
>I guess those 5 misnamed patch files
>are legacy, right?

Probably. I started working on VLC after lobotectomising myself w.r.t. the patchfile naming scheme.

It is quite likely that players like VLC don't work under VirtualBox, esp. not if there still are no "Guest Additions" for Mac.

> @RJVB, I'd squash and commit this, if you don't object, okay?

I haven't checked, did you change anything? If not and it builds for you then no, of course I don't have objections. There's a clear demand for the new version!
